### PR TITLE
Add the ability for star-ratings to have partial stars

### PR DESCRIPTION
**Pull request by stephencookdev**
#### Description

Currently the following

```jsx
import { StarRating } from "baseui/rating";

export default () => {
  return (
    <StarRating value={3.9} />
  );
}
```

will return a component showing 3/5 stars filled.

This is fine, but it's fairly common behaviour for star ratings to show half-stars (i.e. in the example, 3 and a half stars)

This PR adds in support for this, making the example above show 3 and a half stars, rather than a flat 3

#### Scope

Minor: New Feature

### CODE REVIEW

File: src/rating/star-rating.js [hunk 1]
```diff
@@ -79,6 +79,9 @@ class StarRating extends React.Component<StarRatingPropsT, RatingStateT> {
           $isActive={
             previewIndex !== undefined ? x <= previewIndex : x <= value
           }
+          $isPartialActive={
```
Comment (stephencookdev) [inline src/rating/star-rating.js:82]: I can see the argument for making this an opt-in prop. Happy to add if you like. IMO omitting the prop is simpler, and it's fairly unsurprising default behaviour.

But very happy to make this opt-in if people prefer 🙂

**Comment by chasestarr** [comment]
This change LGTM besides the flow error. Can you please run `yarn flow` locally and update?

**Comment by stephencookdev** [comment]
@chasestarr whoops — fixed those now 🙂

**Review by chasestarr — APPROVED** [review]
Looks good, thank you

**Pull request merged**